doc/lua_api.md

```diff
@@ -7532,6 +7532,17 @@ child will follow movement and rotation of that bone.
     * only properties listed in the table will be changed
     * see the 'Object properties' section for details
 * `get_properties()`: returns a table of all object properties
+* `set_observers(observers)`: sets observers (players this object is sent to)
+    * If `observers` is `nil` (or nothing), the object's observers are "unmanaged":
+      The object is sent to all players (in proximity). This is the default.
+    * `observers` is a "set" of player names: `{[player name] = true, [other player name] = true, ...}`
+    * If players are managed, they always need to have themselves as observers.
+    * Attachments:
+        * If an object's observers are managed, the observers of all children need to be managed too.
+        * The observers of children need to be a subset of the observers of parents.
+* `get_observers()`:
+    * returns nothing if the observers are unmanaged
```

Comment:
IMO we shouldn't differentiate nil and nothing in the api docs or usage
```suggestion
    * returns `nil` if the observers are unmanaged
```

Reply:
Should I make the code actually push a `nil` then? I might be pedantic here, but documenting this as `return nil` when it's actually `return` feels wrong to me; there are cases where it could matter.

Reply:
it should push `nil` IMO

Reply:
Done. Also made it throw an error for invalid object refs to avoid surprises.